### ISSUE TITLE
[Bug]: File renaming is not working in webUI, reporting Could not refresh storage stats

### ISSUE THREAD

**Issue by ostasevych** [nextcloud/server#47334]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I try to change or delete some file in webUI it's name is not shown changed or the file deleted until I refresh the page manually. Instead of I see a warning "Could not refresh storage stats". In fact the operation is done.

### Steps to reproduce

1. Pick any filename in webUI.
2. Rename it using the dialogue.
3. Observe that the file name is not renamed, but the warning "Could not refresh storage stats" is shown.
4. Refresh the webpage manually.
5. Observe that the file name is changed.

### Expected behavior

The filename should change immediately in webUI when the operation is done without the webpage refresh.

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 28.0.1 to 28.0.2)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "mycloud.tld",
            "www.mycloud.tld"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "28.0.8.1",
        "overwrite.cli.url": "https:\/\/mycloud.tld\/",
        "htaccess.RewriteBase": "\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_send_plaintext_only": false,
        "default_phone_region": "UA",
        "force_locale": "uk_UA",
        "force_language": "uk",
        "allow_user_to_change_display_name": false,
        "auth.webauthn.enabled": false,
        "remember_login_cookie_lifetime": 1296000,
        "session_lifetime": 21600,
        "session_relaxed_expiry": false,
        "session_keepalive": false,
        "auto_logout": false,
        "skeletondirectory": "\/media\/data\/ncdata\/skeleton",
        "maintenance": false,
        "maintenance_window_start": 23,
        "sort_groups_by_name": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "app_install_overwrite": [
            "files_fulltextsearch",
            "fulltextsearch_elasticsearch",
            "fulltextsearch",
            "files_fulltextsearch_tesseract",
            "twofactor_gateway",
            "files_rightclick",
            "twofactor_reminder"
        ],
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "logtype": "file",
        "logfile": "\/data\/nc\/nextcloud.log",
        "loglevel": 2,
        "loglevel_frontend": 2,
        "log_rotate_size": 104857600,
        "activity_use_cached_mountpoints": true,
        "trashbin_retention_obligation": "auto, 30",
        "logtimezone": "Europe\/Kyiv",
        "filesystem_check_changes": 0,
        "profile.enabled": false,
        "onlyoffice": {
            "verify_peer_off": false,
            "jwt_secret": "***REMOVED SENSITIVE VALUE***",
            "jwt_header": "AuthorizationJwt"
        },
        "enable_previews": true,
        "preview_max_x": "2048",
        "preview_max_y": "2048",
        "enabledPreviewProviders": [
            "OC\\Preview\\Image"
        ],
        "jpeg_quality": "75",
        "preview_concurrency_new": "1",
        "twofactor_enforced": "true",
        "allow_local_remote_servers": true,
        "twofactor_enforced_groups": [],
        "has_internet_connection": true,
        "defaultapp": ""
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - bookmarks: 14.2.4
  - bruteforcesettings: 2.8.0
  - calendar: 4.7.15
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contacts: 5.5.3
  - dav: 1.29.2
  - event_update_notification: 2.4.0
  - external: 5.3.1
  - federatedfilesharing: 1.18.0
  - files: 2.0.0
  - files_accesscontrol: 1.18.1
  - files_automatedtagging: 1.18.0
  - files_fulltextsearch: 28.0.0
  - files_fulltextsearch_tesseract: 27.0.0
  - files_linkeditor: 1.1.20
  - files_pdfviewer: 2.9.0
  - files_scripts: 4.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - fulltextsearch: 28.0.1
  - fulltextsearch_elasticsearch: 28.0.3
  - group_everyone: 0.1.16
  - groupfolders: 16.0.7
  - impersonate: 1.15.0
  - limit_login_to_ip: 4.1.1
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - notes: 4.10.1
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - onlyoffice: 9.3.0
  - otpmanager: 0.5.4
  - password_policy: 1.18.0
  - passwords: 2024.7.20
  - previewgenerator: 5.6.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - quota_warning: 1.19.0
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharelisting: 1.2.0
  - tables: 0.7.5
  - tasks: 0.16.1
  - text: 3.9.2
  - theming: 2.3.0
  - twofactor_admin: 4.5.0
  - twofactor_backupcodes: 1.17.0
  - twofactor_gateway: 0.20.0
  - twofactor_totp: 10.0.0-beta.2
  - updatenotification: 1.18.0
  - viewer: 2.2.0
  - workflowengine: 2.10.0
Disabled:
  - admin_audit: 1.18.0 (installed 1.17.0)
  - circles: 28.0.0 (installed 24.0.0)
  - contactsinteraction: 1.9.0 (installed 1.5.0)
  - dashboard: 7.8.0 (installed 7.4.0)
  - encryption: 2.16.0 (installed 2.15.0)
  - extract: 1.3.6 (installed 1.3.6)
  - federation: 1.18.0 (installed 1.14.0)
  - files_confidential: 3.0.3 (installed 3.0.3)
  - files_downloadactivity: 1.17.0 (installed 1.17.0)
  - files_external: 1.20.0
  - files_lock: 28.0.6 (installed 28.0.6)
  - files_reminders: 1.1.0 (installed 1.0.0)
  - files_rightclick: 0.15.1 (installed 0.15.1)
  - forms: 4.2.4 (installed 4.2.4)
  - metadata: 0.20.0 (installed 0.20.0)
  - nextcloud_announcements: 1.17.0 (installed 1.13.0)
  - photos: 2.4.0 (installed 2.2.0)
  - recommendations: 2.0.0 (installed 1.5.0)
  - related_resources: 1.3.0 (installed 1.0.3)
  - richdocuments: 8.3.9 (installed 8.3.9)
  - secrets: 2.0.3 (installed 2.0.3)
  - sharebymail: 1.18.0 (installed 1.14.0)
  - sharereview: 1.0.2 (installed 1.0.2)
  - support: 1.11.1 (installed 1.7.0)
  - survey_client: 1.16.0 (installed 1.12.0)
  - suspicious_login: 6.0.0
  - systemtags: 1.18.0 (installed 1.18.0)
  - twofactor_reminder: 1.1.0 (installed 1.1.0)
  - user_ldap: 1.19.0
  - user_status: 1.8.1 (installed 1.4.0)
  - weather_status: 1.8.0 (installed 1.4.0)
```

### Nextcloud Signing status

```shell
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- files_downloadactivity
	- INVALID_HASH
		- l10n/uk.js
		- l10n/uk.json
- files_fulltextsearch_tesseract
	- INVALID_HASH
		- templates/settings.admin.php
	- EXTRA_FILE
		- l10n/uk.json
		- l10n/uk.js
- twofactor_gateway
	- INVALID_HASH
		- l10n/uk.js
		- l10n/uk.json
		- lib/Provider/AProvider.php
		- lib/Service/SetupService.php
		- templates/challenge.php
		- lib/Service/Gateway/Signal/Gateway.php
- tasks
	- INVALID_HASH
		- css/tasks-main.css

Raw output
==========
Array
(
    [files_downloadactivity] => Array
        (
            [INVALID_HASH] => Array
                (
                    [l10n/uk.js] => Array
                        (
                            [expected] => c8a50530cda70d2c3b37aed5578e955bb3b968217d18834e76c112ac23ef2c8f03bfb91ce902b55587bbc2ddc8ce4f7f49c7f54f17dccf4b17ce66d30d33f4d1
                            [current] => a67b0cd5cb19cf4b8f7dc7a1e4736b5f486b7b12b3e48de95f4f8f68111166f7b1348746cb54a485e921743b12fa725452336a8c452cc208078cb837387c115d
                        )

                    [l10n/uk.json] => Array
                        (
                            [expected] => 24f688d4a34c63e1ba79094497e9ce3e06793f0031438d4bb218fbf42aee1810db33c00f0574cb6a48cb2ad1bbc8bb4e2c01a8befb93dbdf1ac3deb9d4bbec6d
                            [current] => c6ad2352325864a6c23502052d5cfcfb2a28ebddad052e6c6c3463aa778055679e170f4ced8fb20a730b83bb0c74f310df0e4d497f64476c70ea41495684d203
                        )

                )

        )

    [files_fulltextsearch_tesseract] => Array
        (
            [INVALID_HASH] => Array
                (
                    [templates/settings.admin.php] => Array
                        (
                            [expected] => dbc469abb768cf28595b89a29a68380bd3a72f23aede72c8246ace61d7ac46bb2e45c64f90069083036395f6efae3024595b4b4edc175d43777c2877b1cdbd33
                            [current] => d93d7c672611ff1aa1dd1d8fe08c76a9860b055fc83a1cfd904f54f3f36c453c96cb57861268ff346882f9d404e65b84639426041470a1c52fa3292322a11270
                        )

                )

            [EXTRA_FILE] => Array
                (
                    [l10n/uk.json] => Array
                        (
                            [expected] => 
                            [current] => 1e0fe13577ade092c331c3ad846aadc236da5faaa51512c93e43db4ce79cb6f6042633e0bd4939a3c42447ea8b18078937df4bd3abd7c138ac02873890685179
                        )

                    [l10n/uk.js] => Array
                        (
                            [expected] => 
                            [current] => 4221725a1a2dad708ab385cc03a9069dd4c04060dff6b266abff0db59956197657edb5f1de641ce6d62130ed69598142cd7fc68fdc97a7e3b5652d7906923b6a
                        )

                )

        )

    [twofactor_gateway] => Array
        (
            [INVALID_HASH] => Array
                (
                    [l10n/uk.js] => Array
                        (
                            [expected] => 4f6f8a9535c4f0836139487219a4225b1e3f90a7442ca28ad3d9f3cae0e1185ef68a195b6b82f3d7fb27bf4ec5d24e528dd42b4232022499715cee7c19f61ec5
                            [current] => 5cced5789ee4503090d3a793dd8841aaa84f31983b9cbfb54d0a711b375deeda0b9baf64ccab5e9abbad5a531533bd5806d57a0b723ea27e8d9d930c5d7d785d
                        )

                    [l10n/uk.json] => Array
                        (
                            [expected] => 22bfd148d2af9d55dbc213fb1c54d75755be438709abdff73e8c87c1561b4f1b08d902ef4874bec5b79b207ca62f8d39849bca63e6e18025e73ca9720d8c4316
                            [current] => fc6f766e0acf172b88ae0fd4193ecf79ccf92a08fb2d3573182b2ca5e3072892169d7d7924b31988cdfd7dd2bdd92ff2db729f065835d969bbdcfd169368b221
                        )

                    [lib/Provider/AProvider.php] => Array
                        (
                            [expected] => 7f3cb94751ba4a06c44ffb8ffc0adf940af1dddb917f831a51a6aed2c479e934095c566a71048bccefeb339c265c8c2d40a3b81c7c7fd2bfae53650c53389d90
                            [current] => f87a79743deab95ac36c44a169a1124056f52eb5b16eef92a5a679299eed3c40c6c7928d37af1180b04d2af86b149d9854542b83a74eebafc683fed066a4d351
                        )

                    [lib/Service/SetupService.php] => Array
                        (
                            [expected] => c25ebca2c47f1683c1a10593a236502cef5f8e825441fea2b91b7f1a7b2fac6c42ec8dd9d4841cc9f0facd8bee5b0759c8374b11682cbbc141e900b43e96d9a2
                            [current] => db54c5bb3ce471492b279ee8d4f7ab829bf2c8fbf5d2fb0874037fdda893b8c88cb6d0d34c0997c9cf610e59d0c19f1c39fce34a7d00b534243cd76838703db7
                        )

                    [templates/challenge.php] => Array
                        (
                            [expected] => 72afed33331bfa8344d987a4bade6b5edd8468207bf53da017cc0275d5d6f9b025dadf0e5cc7748acb6ae73a7626d835e014abcc3604cbd5a35eec84cc7dbb6c
                            [current] => 87ed664b9b99e8e6ede154b69607978cecabb82b29f7bfec429a279843169ec2a9cd42725a22f431111fb2d43930b705dc1a632fb5940b758f5b43c91d7ec2be
                        )

                    [lib/Service/Gateway/Signal/Gateway.php] => Array
                        (
                            [expected] => 3127ada11e34a1c33fcc117ffeefc0ab317e1f66312692c21c69c1b6ac82ddc0a5b79aa733eb1d3ad496f746218d1241b1ea9a63e514efe9e0534b8d0529079a
                            [current] => 6778f8c83737992c704e449c70477b0558e86a28cade55149a807d0840feab775865a675504c008ff0cf9276098ec3c17aaebedbdb059cebaa70b7bdaaf856bd
                        )

                )

        )

    [tasks] => Array
        (
            [INVALID_HASH] => Array
                (
                    [css/tasks-main.css] => Array
                        (
                            [expected] => 2127d027d12d9b13a8cfe3f34017853c85b6921c70b0cbd582651729301327e9cf8d5f4a5185adabb6d37a45fe45597d8b636ff76ccd8e9f3fa3ba32cd0b323f
                            [current] => 286029dcc358c13d0259ba0f6c2aef4a97d8d02a7033162a5bc0c97615e712a204cdedf97914ee35c333d40d60f4d7c570fb9245813c75df86bd0f635f30db9a
                        )

                )

        )

)
```

### Nextcloud Logs

```shell
not formatted:
{"reqId":"CQBbi25F0zUm93BYYzcE","level":0,"time":"2024-08-19T23:21:53+03:00","remoteAddr":"**:*:*:****::*","user":"username","app":"no app in context","method":"MOVE","url":"/remote.php/dav/files/username/Documents/test_1231.txt","message":"mount has no provider set, performing full setup","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36","version":"28.0.8.1","data":[],"id":"66c3aaa342848"}

formatted:
[no app in context] Debug: mount has no provider set, performing full setup
	MOVE /remote.php/dav/files/username/Documents/test_1231.txt
	від **:*:*:****::* user username at 19 Aug 2024, 23:21:53

not formatted:
{"reqId":"iQ9AOSobLK2vRq6v0JpW","level":0,"time":"2024-08-19T23:21:54+03:00","remoteAddr":"**:*:*:****::*","user":"username","app":"webdav","method":"PROPFIND","url":"/remote.php/dav/files/username/files/username/Documents/test_12311.txt","message":"File with name //files could not be located","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36","version":"28.0.8.1","exception":{"Exception":"Sabre\\DAV\\Exception\\NotFound","Message":"File with name //files could not be located","Code":0,"Trace":[{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Tree.php","line":95,"function":"getChild","class":"OCA\\DAV\\Connector\\Sabre\\Directory","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":971,"function":"getNodeForPath","class":"Sabre\\DAV\\Tree","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1662,"function":"getPropertiesIteratorForPath","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1647,"function":"writeMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":346,"function":"generateMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPropFind","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/apps/dav/lib/Server.php","line":382,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/www/html/nextcloud/remote.php","line":172,"args":["/www/html/nextcloud/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/www/html/nextcloud/apps/dav/lib/Connector/Sabre/Directory.php","Line":226,"message":"File with name //files could not be located","exception":[],"CustomMessage":"File with name //files could not be located"},"id":"66c3aa6bca06d"}

formatted:
[webdav] Debug: File with name //files could not be located
	PROPFIND /remote.php/dav/files/username/files/username/Documents/test_12311.txt
	від **:*:*:****::* user username at 19 Aug 2024, 23:21:54
```

### Additional info

The warning notice "Could not refresh storage stats" started to appear after moving from php to php-fpm, before it was "File not found", the renaming was not working the same.

**Comment by ostasevych** [comment]
So, the only way to get rid of this issue was to comment out these lines in `config.php`:
```
#  'overwrite.cli.url' => 'https://mycloud.tld/',
#  'htaccess.RewriteBase' => '/',
```

Of course, this lead to losing the ability to hide `index.php` suffix in the URL. Are any other ways to solve the issue?

**Comment by susnux** [comment]
duplicate of https://github.com/nextcloud/server/issues/46604

Fixed in 28.0.9
